python/tk_multi_publish2/dialog.py

```diff
@@ -517,36 +516,66 @@ def _on_item_comment_change(self):
         Callback when someone types in the
         publish comments box in the overview details pane
         """
+        tree_widget = self.ui.items_tree
         comments = self.ui.item_comments.toPlainText()
         # if this is the summary description...
         if self._current_item is None:
             if self._summary_comment != comments:
                 self._summary_comment = comments
 
-                # this is the summary item - so update all top level items and their children!
-                for top_level_item in self._publish_manager.tree.root_item.children:
-                    top_level_item.description = self._summary_comment
-                    for item in top_level_item.descendants:
-                        item.description = comments
-
-                # all tasks have same description now, so set <multiple values> indicator to false
-                self._summary_comment_multiple_values = False
-
-            self.ui.item_comments._show_placeholder = (
-                self._summary_comment_multiple_values
-            )
+                for node_item in tree_widget.root_items():
```

Comment:
I want to make sure I understand this line of code. This visits the UI nodes and not the API nodes, correct, so if you've taken an item and moved it to a separate context even tough it's under a parent in the API hierarchy it is a root in the UI hierarchy. Does what I'm trying to say make any sense to you? Not sure if this is coming out the right way.

Reply:
I'm not sure I fully understand what your asking, however, yes this is traversing the UI nodes, not the API ones.

I'm not sure if I understand the context related bit of your question though. The code is looping over all the collected item nodes. These are sitting under context nodes in the tree (that don't show I think). Perhaps the `tree_widget.root_items` generator should be renamed to `top_level_items`?